CRM/Contribute/Form/Task/Invoice.php

```diff
@@ -471,6 +471,8 @@ public static function printPDF($contribIDs, &$params, $contactIds) {
 
       // condition to check for download PDF Invoice or email Invoice
       if ($isCreatePDF) {
+        // email_comment is an optional variable in this case, and won't usually be present, but allow it to be.
+        $sendTemplateParams['tplParams'] = array_merge($tplParams, ['email_comment' => ($params['email_comment'] ?? '')]);
```

Comment:
This is a really confusing array merge. It feels like a bug waiting to happen for `$sendTemplateParams['tplParams']` to be defined on line 436 and then (maybe) redefined on 475.

Reply:
I just copied it from lines 497 and 512, the only difference being down there it's not optional, so you want the warning if it's missing. I could do a little refactoring, moving all three up to where tplParams is defined and then something like

```php
// If creating a pdf then email_comment is optional, otherwise if it's missing something is wrong.
'email_comment' => $isCreatePDF ? ($params['email_comment'] ?? '') : $params['email_comment'],`
```
That also looks a little weird though.

Reply:
Or, this func probably isn't called from too many places in universe. I could find them all and set $params['email_comment'] explicitly.

Reply:
FWIW I have another PR (currently stale it seems) https://github.com/civicrm/civicrm-core/pull/27162 that is working to get rid of the form variants of 'email_comment' in favour of a generic model property 'userEnteredText' which would be 'notice proof' in all pathways

Reply:
@demeritcowboy yeah I think so - I've updated this field in that PR (ie the invoice template) - but have not test it as yet. Checking to see if tests pass in general

Reply:
I'm not actually sure which tests - the error just appears mixed in with the `...`'s in the phpunit output, but there's several of them appearing so I can probably find at least one.